storagemarket/provider.go

```diff
@@ -202,6 +202,51 @@ func (p *Provider) GetAsk() *storagemarket.StorageAsk {
 	}
 }
 
+func (p *Provider) MakeOfflineDealWithData(dealUuid uuid.UUID, filePath string) (pi *api.ProviderDealRejectionInfo, handler *dealHandler, err error) {
+	p.dealLogger.Infow(dealUuid, "execute offline deal called", "filepath", filePath)
+
+	// db should already have a deal with this uuid as the deal proposal should have been agreed before hand
+	ds, err := p.dealsDB.ByID(p.ctx, dealUuid)
+	if err != nil {
+		return nil, nil, fmt.Errorf("no pre-existing deal proposal for offline deal: %s", err)
+	}
+	if !ds.IsOffline {
+		return nil, nil, errors.New("deal for the given id is not an offline deal")
+	}
+	ds.InboundFilePath = filePath
+
+	// setup clean-up code
+	var dh *dealHandler
+	cleanup := func() {
+		if dh != nil {
+			dh.close()
+			p.delDealHandler(dealUuid)
+		}
```

Comment:
nit: it looks like it's not possible for dh to be nil, so I don't think you need the `if dh != nil`
I'd suggest moving the cleanup function below `dh = p.mkAndInsertDealHandler(dealUuid)` so you don't have to declare the dh variable and initialize it in two separate steps

Reply:
Will do it as part of https://github.com/filecoin-project/boost/pull/338 that will go in after this.